packages/js/test/e2e/global-setup.ts

```diff
@@ -0,0 +1,30 @@
+import { bsLocal, BS_LOCAL_ARGS } from './browserstack.config'
+import { promisify } from 'util'
+
+const sleep = promisify(setTimeout)
+const redColour = '\x1b[31m'
+const whiteColour = '\x1b[0m'
+module.exports = async () => {
+  if (!process.env.BROWSERSTACK_ACCESS_KEY) {
+    console.log('Will not start BrowserStackLocal because BROWSERSTACK_ACCESS_KEY is not set')
+
+    return
+  }
+
+  console.log('Starting BrowserStackLocal ...');
+  // Starts the Local instance with the required arguments
+  let localResponseReceived = false;
+  bsLocal.start(BS_LOCAL_ARGS, (err) => {
+    if (err) {
+      console.error(
+        `${redColour}Error starting BrowserStackLocal${whiteColour}`
+      )
+    } else {
+      console.log('BrowserStackLocal Started')
+    }
+    localResponseReceived = true;
+  });
+  while (!localResponseReceived) {
+    await sleep(1000)
```

Comment:
Calling `sleep` like this is cannot be the right way to do this. But I see that you took it from the example project that browserstack/playwright put out... so not your fault! But yuck 🤦‍♀️ 

Maybe include a comment showing where the example code came from!

https://github.com/browserstack/node-js-playwright-browserstack/blob/main/global-setup.js

Reply:
I decided to improve it :)
No more `sleep`!

Reply:
Thank you! I knew this code must have been copy/pasted from somewhere because your code is much better than this 😆